Makefile

```diff
@@ -142,78 +109,6 @@ image: test ## Build docker image with the manager.
 docker-push: ## Push docker image with the manager.
 	${IMAGE_BUILDER} push ${IMG}
 
-##@ Tests
```

Comment:
Why did you remove these targets? IMO, it's good to have them for development

Reply:
This also affects next commit with github actions modifications. Current approach allows us to use exactly the same commands during development and CI processes

Reply:
To begin with these lines are largely the reasons used for justification for the `$(BASE)` and `GOPATH` hackery, so I've removed them to make the `Makefile` clean.

Also the targets/aliases in question are almost entirely unnecessary/unused:
- `lint`: `golangci-lint` it's in the name, it's a linter for CI. Your own go development environment will have a linter built in.
- `lint-dockerfile`: The `Dockerfile` is static and doesn't need a linter, but I still added it to the GitHub actions. 
- `helm lint`: Might be something to retain.
- `test-bench`: Alias target for passing an argument to the test command, never seen this used. What does this provide?
- `test-short`: Alias target for passing an argument to the test command, never seen this used. Why would one want to run only a subset of tests? The full `test` target with verbosity and coverage takes ~15s to run.
- `test-verbose`: Alias target for passing an argument to the test command. This is the default I have set for the `test` target.
- `check test tests`: Very curious why one would want to have three test aliases for a single test target.
- `test-xml`: Unused, `GO2XUNIT` not defined in the `Makefile` and the tool is depricated.
- `test-coverage`: Coverage output as part of new `test` target.
- `test-coverage-tools`: Unused, uploads coverage output to Coveralls.io.

Reply:
I agree about test-* targets but I still want to keep *lint targets as is.